### ISSUE TITLE
Improve resistor-color-duo: detect string concat

### ISSUE THREAD

**Issue by SleeplessByte** [exercism/javascript-analyzer#55]
**Describe the improvement**:

When the following main method is provided, the analyzer bails early:

```javascript
export const value = arr => {
  return +(COLORS.indexOf(arr[0]) + "" + COLORS.indexOf(arr[1]));
};
```

Instead, it should be caught by `hasDigitsString`, and produce the same comment.

**Which exercise**

`resistor-color-duo`

**Additional context**

The `ResistorColorDuoSolution` has a method that checks if an exercise _has a constructed string_. It currently only detects a string template literal that looks like `${a}${b}`, but we want to also capture the following:

```javascript
`${a}` + `${b}`
a + "" + b
a + '' + b
""  + a + b
''  + a + b
a.toString() + b
a.toString() + b.toString()
a.toString().concat(b)
```
